clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/PartitionConsumptionState.java

```diff
@@ -442,6 +443,10 @@ public void maybeUpdateExpectedChecksum(byte[] key, Put put) {
     if (this.expectedSSTFileChecksum == null) {
       return;
     }
+    // Checksum calculation should skip RMD chunk.
+    if (put.schemaId == AvroProtocolDefinition.CHUNK.getCurrentProtocolVersion() && put.putValue.remaining() == 0) {
```

Comment:
I re-read the code, so if chunking is disabled, the RMD payload will always be in the same message of `put` containing the put value payload.
So this might not be an issue.
Feel free to ignore this if you confirm this is a false alarm.

Reply:
RMD chunking has two parts:
chunks and manifest,  and we only ignore the chunks here, what about the chunk manifest for RMD payload, does it matter?

Reply:
According to our original design, the chunk manifest of RMD is in the last message's rmd payload, so it does not matter.
It is like:
value_chunk_1, ..., value_chunk_m, rmd_chunk_1, ..., rmd_chunk_n, manifest(put payload: value manifest, rmd payload: rmd manifest).

Reply:
Yeah, and with no-chunked case, what you said in the 2nd reply is correct. So I don't think we should itertate on that. In theory, 2nd condition might be enough, but the 1st condition makes it clear it is a chunk. So it would not impact any future changes.

Reply:
Let us update the javadoc to include these since it is a little confusing at least when I firstly look at this change.

Reply:
Sure, just updated.

Reply:
Nitpick... can we merge the two comments into the same comment block?